### PR TITLE
Extract dropdown

### LINKED ISSUE #3418: Filter PopOver goes off screen in Mobile devices (sm and small under 450px)
### Package

filament/tables

### Package Version

2.15.3

### Laravel Version

9.22.1

### Livewire Version

2.10

### PHP Version

8.1.4

### Problem description

On mobile screens, the Popover filter div is going off-screen. Most likely caused by the left margin.

### Expected behavior

The popOver filter must not overflow

### Steps to reproduce

1. Attached is a simple project
2. Connect SQLite database
3. Run the project and check the home page in mobile view
4. Try to filter 

### Reproduction repository

https://github.com/lucidpolygon/filament-asda

### Relevant log 
Screen Shot 2022-08-06 at 5 16 43 PM
output

_No response_

### PR DESCRIPTION
**Pull request by zepfietje**
- [x] Update docs about configuring scripts
- [x] Dropdown panel width
- [x] Dropdown border opacity

### CODE REVIEW

**Comment by jojostx** [comment]
Hello @zepfietje, great work!!. 

I have a suggestion related to the drop-down functionality. It is somewhat simple and may not need me to open an issue for it.

The Table ActionGroup drop-down currently uses a hard-coded value of 'w-52' which causes a truncation of items in the drop-down eg: 
Screenshot (18)

If you could allow customization of the width for the action group table component similar to how modals allow customizing the width, it would fix the issue. 
eg: 
for current implementation for Modals
Screenshot (20)

proposed usage of width customization for ActionGroup:
Screenshot (24)

**Comment by zepfietje** [comment]
Thanks for your detailed explanation, @jojostx!

I will have a look at what works best for the dropdown size and take your remarks into account.

File: packages/forms/docs/01-installation.md [hunk 1]
```diff
@@ -39,14 +39,15 @@ You're now ready to start [building forms](getting-started)!
...
+- [Alpine Floating UI](https://github.com/awcodes/alpine-floating-ui)
 - [Tailwind CSS](https://tailwindcss.com)
 - [Tailwind CSS Forms plugin](https://github.com/tailwindlabs/tailwindcss-forms)
 - [Tailwind CSS Typography plugin](https://tailwindcss.com/docs/typography-plugin)
 
 You may install these through NPM:
 
 ```bash
-npm install alpinejs tailwindcss @tailwindcss/forms @tailwindcss/typography --save-dev
+npm install alpinejs @awcodes/alpine-floating-ui tailwindcss @tailwindcss/forms @tailwindcss/typography --save-dev
```
Comment (danharrin) [inline packages/forms/docs/01-installation.md:50]: Breaking change?

Comment (zepfietje) [reply packages/forms/docs/01-installation.md:50]: Yeah, just changed it to how we'd ideally have it for now.
We'll have to discuss how we'd like to actually handle this before merging the PR...

**Comment by danharrin** [comment]
As per #2236, we should replace the dropdown border with a ring

**Comment by zepfietje** [comment]
@danharrin, keeping the border but modifying the border color opacity would yield the same result, right?

I've added it to the list of tasks to be done and will have a look at what works best. 👌

File: packages/tables/resources/views/components/filters/popover.blade.php [hunk 2]
```diff
@@ -3,54 +3,18 @@
...
-                '7xl' => 'max-w-7xl',
-                default => 'max-w-sm',
-            },
-        ])
-    >
-        <div @class([
-            'px-6 py-4 bg-white border border-gray-300 space-y-6 shadow-xl rounded-xl',
-            'dark:bg-gray-800 dark:border-gray-700' => config('tables.dark_mode'),
-        ]) wire:ignore.self>
-            <x-tables::icon-button
```
Comment (zepfietje) [inline packages/tables/resources/views/components/filters/popover.blade.php:45]: @danharrin, I've removed the filter dropdown close button.

1. It's not necessary.
2. It bloats the UI.
3. It's inconsistent with other dropdowns (why would the columns toggle panel have no close button?).

**Comment by danharrin** [comment]
Merge conflicts

**Comment by zepfietje** [comment]
> Merge conflicts

Resolved!

File: packages/forms/docs/01-installation.md (hunk 1 above)
Comment (zepfietje) [reply packages/forms/docs/01-installation.md:50]: So what would you suggest, @danharrin? Importing Alpine Floating UI inside the forms package results in duplicate bundled code if you also manually import it elsewhere.

Comment (danharrin) [reply packages/forms/docs/01-installation.md:50]: It needs to stay in the forms package until v3

**Review by danharrin — CHANGES_REQUESTED** [review]
You will need to revert all asset changes before I can merge this

**Comment by zepfietje** [comment]
> You will need to revert all asset changes before I can merge this

Yeah, so want to keep Alpine Floating UI shipped with the forms package? And what about the others, should they have docs to manually add it, since it wasn't there now?

**Comment by danharrin** [comment]
Floating UI is used by the forms package

Tables and admin require the forms package already

So no extra docs needed, just keep this PR as a refactor to a common dropdown Blade component.

**Comment by zepfietje** [comment]
I've got a separate PR to add action groups support to the notifications package. So how should that be handled then?

**Comment by zepfietje** [comment]
> You will need to revert all asset changes before I can merge this

Resolved, @danharrin!

File: packages/support/resources/views/components/dropdown/index.blade.php [hunk 3]
```diff
@@ -21,7 +21,7 @@
     }"
 >
     <div
-        class="filament-dropdown-trigger"
+        class="filament-dropdown-trigger cursor-pointer"
```
Comment (zepfietje) [inline packages/support/resources/views/components/dropdown/index.blade.php:24]: @danharrin I left this out on purpose: https://github.com/tailwindlabs/tailwindcss/pull/8962.

Comment (danharrin) [reply packages/support/resources/views/components/dropdown/index.blade.php:24]: The rest of our buttons have that behaviour so I added it for consistency.

Comment (zepfietje) [reply packages/support/resources/views/components/dropdown/index.blade.php:24]: Understand. Let's discuss this for v3?

**Pull request merged**